Assets/MixedRealityToolkit/_Core/Interfaces/Sharing/ISimpleState.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces.Sharing
+{
+    /// <summary>
+    /// Simplest possible non-automatic state sharing.
+    /// Will typically be networked but could also plug into a file system or local database.
+    /// </summary>
+    public interface ISimpleState : IMixedRealityService
+    {
+        Task<bool> HasKey(string key);
+        Task SetState<T>(string key, T state);
```

Comment:
Do we have a perspective on how multiple asynchronous sets should behave? It feels like there is the possibility for a set to fail, which leads me to think this should return a Task<bool> compared to just a Task?

Reply:
Fair point but I have a feeling it won't be used in practice. Here's a straightforward case:
```
await simpleState.SetKey("Object1.Position", transform.position);
await simpleState.SetKey("Object1.Rotation", transform.eulerAngles);
await simpleState.SetKey("Object1.Active", gameObject.activeSelf);
...
```
This is awkward by contrast. I have a feeling devs would opt to plow ahead and ignore failed set operations:
```
if (!simpleState.SetKey("Object1.Position", transform.position).Result)
    throw new Exception("Failed to set key Object1.Position"); // or handle some other way

if (!simpleState.SetKey("Object1.Rotation", transform.eulerAngles).Result)
    throw new Exception("Failed to set key Object1.Rotation");

if (!simpleState.SetKey("Object1.Active", gameObject.activeSelf).Result)
    throw new Exception("Failed to set key Object1.Active");
```
If we assume serialization issues will be caught at compile time (big if) then connection errors will be where 99% of problems arise. Maybe an IsConnected property would give them enough to bail early?

```
if (!simpleState.IsConnected)
    return;

await simpleState.SetKey("Object1.Position", transform.position);
...
```